qiskit/circuit/quantumcircuit.py

```diff
@@ -824,8 +826,10 @@ def add_register(self, *regs):
                                    % register.name)
             if isinstance(register, QuantumRegister):
                 self.qregs.append(register)
+                self._qubits.extend(register)
```

Comment:
does this need to be `extend(register[:])`?

Reply:
`extend` can work directly from `Register.__iter__`. `register[:]` would work too, if you think that's clearer.